lib/pymedphys/_experimental/streamlit/apps/extend_ct.py

```diff
@@ -0,0 +1,251 @@
+
+    st.write(
+        """
+        ## Select patient
+
+        For a patient to show up below, make sure to run a DICOM export
+        to file within Monaco first.
+        """
+    )
+    chosen_patient_id = st.radio("Patient ID", patient_ids)
+
+    ct_dicom_files = list(dicom_export.glob(f"{chosen_patient_id}_*_image*.DCM"))
+    _stop_button_if_cache_miss("Load files", _load_dicom_files, ct_dicom_files)
+
+    ct_datasets = _cached_load_dicom_files(ct_dicom_files)
+    patient_names_found = {header.PatientName for header in ct_datasets}
+    st.write(patient_names_found)
+
+    if len(patient_names_found) != 1:
+        raise ValueError("Expected exactly one patient name to be found.")
+
+    st.write(
+        """
+        ## Extension of CT scan
+        """
+    )
+
+    chosen_number_of_slices = st.number_input(
+        "Number of slices to extend by (both sup and inf)", min_value=0, value=20
+    )
+
+    if not st.button("Extend CT and send back to Monaco"):
+        st.stop()
+
+    extended_ct_datasets = _extend_datasets(ct_datasets, chosen_number_of_slices)
+    _send_datasets(hostname, port, extended_ct_datasets)
+
+    st.success(
+        """
+        The CT expansion and DICOM send to Monaco process was successful.
+        Now use the import new DICOM files functionality within Monaco
+        to retrieve the new extended dataset.
+        """
+    )
+
+
+def _send_datasets(hostname, port, datasets):
+    status = st.empty()
+    progress_bar = st.progress(0)
+
+    total_number_of_datasets = len(datasets)
+
+    with association(hostname, port) as assoc:
+        for i, ds in enumerate(datasets):
+            ds.TransferSyntaxUID = pydicom.uid.ExplicitVRLittleEndian
+            ds.fix_meta_info(enforce_standard=True)
+
+            returned_status = assoc.send_c_store(ds)
+            readable_status = pynetdicom.status.code_to_category(returned_status.Status)
+            slice_location = ds.SliceLocation
+
+            status.write(
+                f"Slice Location: {slice_location} | Status: {readable_status}"
+            )
+
+            progress_bar.progress((i + 1) / total_number_of_datasets)
+
+
+@contextlib.contextmanager
+def association(hostname, port):
+    try:
+        assoc = _get_association(hostname, port)
+        yield assoc
+    finally:
+        assoc.release()
+
+
+def _get_association(hostname, port):
+    ae = _get_ae()
+    assoc = ae.associate(hostname, int(port))
+
+    return assoc
+
+
+@st.cache(allow_output_mutation=True)
+def _get_ae():
+    ae = pynetdicom.AE()
+    ae.network_timeout = None
+    ae.acse_timeout = None
+    ae.dimse_timeout = None
+    ae.maximum_pdu_size = 0
+
+    ae.add_requested_context(
+        pynetdicom.sop_class.CTImageStorage  # pylint: disable=no-member
+    )
+
+    return ae
+
+
+def _extend_datasets(datasets, number_of_slices):
+    deque_datasets = _extend.convert_datasets_to_deque(datasets)
+    _extend.extend_datasets(deque_datasets, 0, number_of_slices)
```

Comment:
What does this do?
I'm  on my phone and searching for the key function in a different place is awkward.  Does it duplicate the slice end but alter the image position patient by the difference of the previous slice (add it to the stack).
This is something people frequently get wrong because of assumptions

Reply:
This function being called is code from 2019 which is within the following:

https://github.com/pymedphys/pymedphys/blob/main/lib/pymedphys/_dicom/ct/extend.py

with tests over at:

https://github.com/pymedphys/pymedphys/blob/main/lib/pymedphys/tests/dicom/test_extend_ct.py

It's heavily dependent on the `SliceLocation` parameter. And I have suspicions that there is a range of cases not accounted for. If that is the case I would prefer to test for those cases and raise errors, as opposed to supporting them at this time.

Reply:
Alright, so this PR now includes an additional `require_patient_orientation(ct_series, "HFS")` which should cover the cases I could envisage where this extend CT code will fail mightily.

Reply:
Patient Position (0018,5100) is independent of Image Orientation Patient.
I have seen CT where the Image Orientation Patient was flipped around instead of the pixels (old Picker CT), so if Image Orientation Patient isn't what you would expect given Patient Positionn, things can get ugly.
I haven't seen that happen with HFS though, so given there aren't too many Picker CT floating around anymore...
(This isn't a ding on Picker... they had a right to do that... it just made life a lot harder for me...)

Reply:
> Patient Position (0018,5100) is independent of Image Orientation Patient.

Yup, I shall make sure to verify both.